snakemake/executors.py

```diff
@@ -2052,3 +2059,605 @@ def run_wrapper(
             write_benchmark_records(bench_records, benchmark)
         except (Exception, BaseException) as ex:
             raise WorkflowError(ex)
+
+
+GoogleLifeScienceJob = namedtuple(
+    "GoogleLifeScienceJob", "job jobname jobid callback error_callback"
+)
+
+
+class GoogleLifeScienceExecutor(ClusterExecutor):
+    """the GoogleLifeSciences executor uses Google Cloud Storage, and
+       Compute Engine paired with the Google Life Sciences API.
+       https://cloud.google.com/life-sciences/docs/quickstart
+    """
+
+    def __init__(
+        self,
+        workflow,
+        dag,
+        cores,
+        envvars,
+        jobname="snakejob.{name}.{jobid}.sh",
+        printreason=False,
+        quiet=False,
+        printshellcmds=False,
+        container_image=None,
+        regions=None,
+        cache=False,
+        machine_type_prefix=None,
+        latency_wait=3,
+        local_input=None,
+        restart_times=None,
+        exec_job=None,
+        max_status_checks_per_second=1,
+    ):
+
+        # Attach variables for easy access
+        self.workflow = workflow
+        self.quiet = quiet
+        self.workdir = os.path.dirname(self.workflow.persistence.path)
```

Comment:
This is the subfolder `.snakemake` inside of the working directory. Is that intended?

Reply:
Ok, now I understand what you do here. You can instead just use `os.getcwd()`